mir_eval/util.py

```diff
@@ -956,3 +957,79 @@ def midi_to_hz(midi):
         Frequency/frequencies in Hz corresponding to `midi`
     '''
     return 440.0 * (2.0 ** ((midi - 69.0)/12.0))
+
+
+def estimate_hop_length(times):
+    """Estimate hop length of a semi-regular but non-uniform series of times.
+
+    Parameters
+    ----------
+    times : ndarray
+        Array of times corresponding to a time series
+
+    Returns
+    -------
+    hop_length : number
+        Estimated hop length (seconds)
+    """
+
+    # Make sure the times are sorted
+    times = np.sort(times)
+
+    # Estimate the hop in seconds
+    non_gaps = np.append([False], np.isclose(np.diff(times, n=2), 0))
+
+    if not np.sum(non_gaps):
+        raise ValueError("Time observations are too irregular.")
+
+    return np.diff(times)[non_gaps].mean()
```

Comment:
I think `.median()` would be more stable, since in theory the spacing should be uniform. Then one or two outliers won't skew the hop size, which will over time propagate the error